src/output-json-alert.c

```diff
@@ -93,12 +93,13 @@ typedef struct JsonAlertLogThread_ {
 
 /* Callback function to pack payload contents from a stream into a buffer
  * so we can report them in JSON output. */
-static int AlertJsonPrintStreamSegmentCallback(const Packet *p, void *data, uint8_t *buf, uint32_t buflen)
+static int AlertJsonDumpStreamSegmentCallback(const Packet *p, void *data, uint8_t *buf, uint32_t buflen)
 {
     MemBuffer *payload = (MemBuffer *)data;
 
-    PrintStringsToBuffer(payload->buffer, &payload->offset, payload->size,
-                         buf, buflen);
+    // We can't user Print...ToBuffer functions as we need real raw data.
+    memcpy(payload->buffer + payload->offset, buf, buflen);
```

Comment:
I think it'll be better to use payload->size instead of buflen. Payload is 4k, but who knows if it's enough.